### PR TITLE
Moving constants to firebase

### PR DESCRIPTION
**Pull request by charlesx2013**
^ @SamLau95

### CODE REVIEW

File: src/js/chat.js [hunk 1]
```diff
@@ -246,25 +251,36 @@ class Room extends AbstractRoom {
   }
 }
 
+const CONST_FIREBASE = new Firebase(`https://research-chat-room.firebaseio.com/constants`);
```
Comment (SamLau95) [inline src/js/chat.js:254]: If this is a constant, this should go in `constants.js`, right?

**Comment by SamLau95** [comment]
I actually don't really like the idea of it technically being possible to have a state where the constants aren't initialized yet but the chat room tries to move on. For example, it's possible that only the Firebase with constants time-outs. Then, what happens? The user gets sent to the waiting room, and we try to set a timer so that if the user waits 3 minutes without finding a match he/she gets moved on to the next part of the survey. However, that timer length will be undefined since the Firebase didn't load the variable. I think we should ensure that nothing happens before we read in the settings from Firebase. This cuts down on the amount of state we'll have to think about.

Implementation-wise, maybe we should rename `constants.js` to `settings.js` and load all the settings then, blocking progress until they're ready to go.

**Comment by SamLau95** [comment]
Ping @charlesx2013 ; we're having our midpoint check-in with Angus in the next week or so.

**Comment by SamLau95** [comment]
@charlesx2013 ETA on this?

**Comment by charlesx2013** [comment]
I'll work on it tonight.

**Pull request merged**